### PR TITLE
Update version of ICU data via icuexportdata to release-72-1

### LINKED ISSUE #2770: Update icuexportdata to version release-72-1 to get Unicode 15 data
## Background

The data we get from ICU via `icuexportdata` contains compactly packaged versions of data from Unicode sources for properties, collations, etc.

Currently, ICU4X 1.0 is using the `icu4x/2022-08-17/71.x` release tag of ICU as the snapshot from which to download and use `icuexportdata`.  This tag is a git ref to a commit on the ICU `maint/maint-71` branch because ICU 71 code uses Unicode 14 data, against which ICU4X development was done leading up to 1.0.

ICU 72 was released yesterday, with a relase tag of `release-72-1`.  It contains data for Unicode 15.0.

## Issue

We need to make changes in ICU4X to update to Unicode 15.0 via `icuexportdata` in a couple of ways:

- [x] #1921
- [ ] Update the git tag of the release used for `icuexportdata`.  (Note: PR #2371 is an example that does this.)
- [ ] Update collation test data files used in collator tests (test data files are kept in sync with collation sort key data per version)

For updating the release git tag of `icuexportdata`, I currently see a few places where this should happen:

* CI workflow `build-test.yml` 
  * References directly to the tag (ex: `"release-72-1"`)
  * Filenames that encode the tag name within it (ex: `"icuexportdata_release-72-1.zip"`), in which any slashes in the tag become dashes
* `provider/testdata/Cargo.toml` - a field called `icuexportdata_gitref`
* `provider/testdata/src/lib.rs` - a doc test for `icu_tag()` asserting the git ref string
* `ffi/diplomat/js/examples/node/build.sh` - git ref string when running `icu_datagen` to create dat afor the WASM demo's `.postcard` data

When working on the PR branch locally, remember to run `cargo make testdata-download-sources` to update the test data accordingly.

### PR DESCRIPTION
**Pull request by echeran**
Fixes #2770, #2764 

Is blocked on issue #1921 being fixed first.

### CODE REVIEW

**Comment by echeran** [comment]
@hsivonen Would you be able to help me figure out why the tests are failing in this PR?  There are a few tests similar to [this one](https://github.com/unicode-org/icu4x/actions/runs/3325134303/jobs/5497484381#step:7:982) that give similar panic stacktraces:

```
thread 'transform::icuexport::collator::test::test_fi' panicked at 'called `Result::unwrap()` on an `Err` value: Data(DataError { kind: MissingLocale, key: Some(DataKey{collator/meta@1}), str_context: None })', provider/datagen/src/transform/icuexport/collator/test.rs:22:83
stack backtrace:
...
```

Note that this PR is try to both update the git tag for icuexportdata to `release-72-1` and also remove the temporary workaround for the `sv` locale file naming from PR #2765.  The reason for combining the workaround removal into the icuexportdata version update is that the test that verifies that testdata is consistent was failing when I only updated the icuexportdata version.

**Comment by hsivonen** [comment]
> @hsivonen Would you be able to help me figure out why the tests are failing in this PR?

The provider formats the file path as `"collation/implicithan/fi_meta.toml"` in https://github.com/unicode-org/icu4x/blob/6bf69ba0a5bb6504a18a95829d3ffdbbca414a51/provider/datagen/src/transform/icuexport/collator/mod.rs#L127 while the file on disk is named `collation/implicithan/fi_standard_meta.toml`.

**Comment by hsivonen** [comment]
Looks like the problem is that this PR changes these lines:
https://github.com/unicode-org/icu4x/blob/6bf69ba0a5bb6504a18a95829d3ffdbbca414a51/provider/datagen/src/transform/icuexport/collator/mod.rs#L66-L80

**Comment by hsivonen** [comment]
`git revert c3933c805abedb7097eaf019b4bd2c140cc6b816` fixes this for me.

Then, s/sv_reformed/sv_traditional/ in provider/testdata/Cargo.toml would be proper but doesn't actually affect the test result.

**Comment by echeran** [comment]
@hsivonen I applied those changes, and applied the output changes from `cargo make testdata`, and fixed a tag update for the full-datagen test. Now I'm getting what I think is just a failing unit test.

When I run `cargo test -p icu_collator --test tests`, I get:

```
---- test_conformance_non_ignorable stdout ----
thread 'test_conformance_non_ignorable' panicked at 'assertion failed: `(left == right)`
  left: `"\u{f83}\u{334}"`,
 right: `"\u{334}\u{f86}"`', components/collator/tests/tests.rs:1400:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_conformance_shifted stdout ----
thread 'test_conformance_shifted' panicked at 'assertion failed: `(left == right)`
  left: `"\u{f83}!"`,
 right: `"\u{f86}!"`', components/collator/tests/tests.rs:1355:17
```

Is that due to something I did? Or did the data change (I'm guessing not)?  For reference, here are the Unicode 15.0 values for [U+0F83](https://util.unicode.org/UnicodeJsps/character.jsp?a=0f83&B1=Show) and [U+0F86](https://util.unicode.org/UnicodeJsps/character.jsp?a=0f86&B1=Show)

**Comment by hsivonen** [comment]
The tests pass if I [update the test data to latest as well](https://github.com/echeran/icu4x/pull/30).

So at least the new test data and the new collation root are self-consistent. I didn't investigate why the collation order changed for characters that are assigned so many Unicode versions ago (assigned in Unicode 2.0, AFAICT).

**Comment by hsivonen** [comment]
https://www.unicode.org/Public/UCA/14.0.0/allkeys.txt had:
`0F83  ; [.0000.0000.0000] # TIBETAN SIGN SNA LDAN`

https://www.unicode.org/Public/UCA/15.0.0/allkeys.txt has:
0F83  ; [.0000.00C3.0002] # TIBETAN SIGN SNA LDAN

**Comment by hsivonen** [comment]
I suspect this is a side effect of https://www.unicode.org/L2/L2022/22128-script-adhoc-rept.pdf (page 27)

@markusicu, do I understand correctly that the root collation change for U+0F83 is intentional?

File: ffi/diplomat/js/examples/node/build.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ if ! test -f "full-data-cached.postcard"; then
         --all-locales \
         --all-keys \
         --cldr-tag 41.0.0 \
-        --icuexport-tag icu4x/2022-08-17/71.x \
+        --icuexport-tag release-72-1 \
```
Comment (sffc) [inline ffi/diplomat/js/examples/node/build.sh:32]: Thought: Maybe we should keep this one on the old data version? But I guess we want the WASM demo to be using the latest data. https://github.com/unicode-org/icu4x/issues/2782#issuecomment-1300991060

**Review by hsivonen — APPROVED** [review]
LGTM provided that @sffc 's points are addressed. Thanks.

File: components/collator/tests/data/CollationTest_CLDR_NON_IGNORABLE.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # CollationTest_CLDR_NON_IGNORABLE_SHORT.txt
```
[the remark this answers is not in the document]
Comment (hsivonen) [reply components/collator/tests/data/CollationTest_CLDR_NON_IGNORABLE.txt:1]: (I also ran the tests with the full-length test data.)

File: provider/testdata/Cargo.toml [hunk 3]
```diff
@@ -166,8 +166,8 @@ icuexportdata_glob = [
     "collation/implicithan/root_standard_jamo.toml",
     "collation/implicithan/root_standard_meta.toml",
     "collation/implicithan/root_standard_prim.toml",
-    "collation/implicithan/sv_reformed_data.toml",
-    "collation/implicithan/sv_reformed_meta.toml",
+    "collation/implicithan/sv_traditional_data.toml",
+    "collation/implicithan/sv_traditional_meta.toml",
```
[the remark this answers is not in the document]
Comment (echeran) [reply provider/testdata/Cargo.toml]: Done. Oops, a bit ironic, too.

File: provider/datagen/src/source.rs [hunk 4]
```diff
@@ -94,7 +94,7 @@ impl SourceData {
     /// using the given tag. (see [GitHub releases](https://github.com/unicode-org/icu/releases)).
     pub fn with_icuexport_for_tag(self, mut tag: &str) -> Result<Self, DataError> {
         if tag == "release-71-1" {
-            tag = "icu4x/2022-08-17/71.x";
+            tag = "release-72-1";
         }
```
[the remark this answers is not in the document]
Comment (echeran) [reply provider/datagen/src/source.rs]: Done.

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**